### DISCUSSION TITLE
Could not determine IPv4/IPv6 protocol

### DISCUSSION THREAD

**Discussion by adamh98** [haugene/docker-transmission-openvpn#2316] [Q&A]
2022-08-12 23:08:50 Could not determine IPv4/IPv6 protocol
2022-08-12 23:08:50 SIGUSR1[soft,init_instance] received, process restarting
2022-08-12 23:08:50 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
2022-08-12 23:08:50 RESOLVE: Cannot resolve host address: {redacted} (Name does not resolve)
2022-08-12 23:08:50 RESOLVE: Cannot resolve host address: {redacted} (Name does not resolve)
2022-08-12 23:08:50 Could not determine IPv4/IPv6 protocol
2022-08-12 23:08:50 SIGUSR1[soft,init_instance] received, process restarting

I am adding the dns flag near the beginning of the command with 8.8.8.8. 

I'm able to get into the container and resolve goolge or PIA. I can see in the resolv.conf file, the nameserver is correct. I've adjusted it to PIA's recommended servers as well with no luck.

It was working just fine until I had to restart my machine and the container stopped working. I've now explicitly tried the build 3.7.1

If someone could help me out that would be great.

**Comment by pkishino** [comment]
Have you tried :dev branch?

**Reply by adamh98** [reply]
I have not. How would I go about doing that?

**Reply by pkishino** [reply]
instead of pulling the 3.7.1 build pull from dev instead
docker pull haugene/docker-transmission-openvpn:dev

**Reply by adamh98** [reply]
Got it. I'll give that a shot and see. Thanks for the suggestion.

**Reply by adamh98** [reply]
I gave it a try and unfortunately, the same result as before.

**Reply by pkishino** [reply]
Well, can’t help more without your config etc..
Post the whole config with all details except password and full log please

**Reply by adamh98** [reply]
docker run --cap-add=NET_ADMIN -d \
              --dns 8.8.8.8 \
              --dns 8.8.4.4 \ 
              -v /media/nfs/:/data \
              -e OPENVPN_PROVIDER=PIA \
              -e OPENVPN_CONFIG=us west \
              -e OPENVPN_USERNAME={user} \
              -e OPENVPN_PASSWORD={pass} \
              -e "OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60" \
              -e LOCAL_NETWORK=10.20.30.0/24 \
              -e TRANSMISSION_RPC_USERNAME={username} \ 
              -e TRANSMISSION_RPC_PASSWORD={password} \
              --log-driver json-file \
              --log-opt max-size=10m \
              -p 9091:9091 \
              haugene/transmission-openvpn:dev

**Reply by adamh98** [reply]
Executing setup script for PIA
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.BsZc68m9tA
Extract OpenVPN config bundle into PIA directory /etc/openvpn/pia
Starting OpenVPN using config us_west.ovpn
Modifying /etc/openvpn/pia/us_west.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Updating status for config failure detection
Setting OpenVPN credentials...
adding route to local network 10.20.30.0/24 via 172.17.0.1 dev eth0
Mon Aug 15 15:23:40 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Mar 22 2022
Mon Aug 15 15:23:40 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Mon Aug 15 15:23:40 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Mon Aug 15 15:23:40 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 15:23:40 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 15:23:45 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 15:23:50 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 15:23:50 2022 Could not determine IPv4/IPv6 protocol
Mon Aug 15 15:23:50 2022 SIGTERM[soft,init_instance] received, process exiting
Starting container with revision: c06a8ad89068a1e7a68f889dbc2351e86c107f1a
Creating TUN device /dev/net/tun
Using OpenVPN provider: PIA
Running with VPN_CONFIG_SOURCE auto
Provider PIA has a bundled setup script. Defaulting to internal config
Executing setup script for PIA
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.9wO5W7n5M6
Extract OpenVPN config bundle into PIA directory /etc/openvpn/pia
Starting OpenVPN using config us_west.ovpn
Modifying /etc/openvpn/pia/us_west.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Updating status for config failure detection
Setting OpenVPN credentials...
adding route to local network 10.20.30.0/24 via 172.17.0.1 dev eth0
Mon Aug 15 23:22:45 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Mar 22 2022
Mon Aug 15 23:22:45 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Mon Aug 15 23:22:45 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Mon Aug 15 23:22:45 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 23:22:45 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 23:22:50 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name or service not known)
Mon Aug 15 23:23:58 2022 Could not determine IPv4/IPv6 protocol
Mon Aug 15 23:23:58 2022 SIGTERM[soft,init_instance] received, process exiting

**Reply by pkishino** [reply]
first of all,please check this:
docker run --cap-add=NET_ADMIN -d
-v /your/storage/path/:/data
--dns 8.8.8.8
--dns 8.8.4.4 \
-v /media/nfs/:/data

you are specifying two different locations for /data..
second, please check dns settings etc for your docker host as well

**Reply by adamh98** [reply]
Thank you for your help. I forgot to remove that when I copied it for the reply. That’s not in my config. I’ll double check the host and make sure. It was good the last time I saw it.

**Reply by adamh98** [reply]
I looked on the host dns settings, and the 8.8.8.8 and 8.8.4.4 are present. I'm able to ping privateinternetaccess.com and nslookup the site. 

I'm not sure what could be causing it since I can resolve the address within the container and outside.

**Reply by pkishino** [reply]
can you try remove the dns in the container config please? and try a different server..
somehow it is not able to contact the pia servers at all.
btw, considering the container would be continuously restarting, how did you test the connection inside the container?

**Reply by adamh98** [reply]
I'll give that a try. I'll put the PIA DNS servers in and try it. Thank you.

Before declaring the dev pull, I was able to get into the container and the connections were restarting. The logs showed the cycle. I had time to go through and check the container configs. Now, I'm not able to do that.

**Reply by adamh98** [reply]
_Inside Container_

bash-5.0# cat resolv.conf
nameserver 209.222.18.218
nameserver 209.222.18.222
bash-5.0# exit

_Container Logs_
admin@linux1:~$ sudo docker logs d8
Starting container with revision: a4d65774f855a04070766e53346661f48c76fa0e
Creating TUN device /dev/net/tun
Using OpenVPN provider: PIA
Provider PIA has a custom setup script, executing it
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.POpFnl
Extract OpenVPN config bundle into PIA directory /etc/openvpn/pia
Starting OpenVPN using config us_west.ovpn
Modifying /etc/openvpn/pia/us_west.ovpn for best behaviour in this container
Setting OpenVPN credentials...
adding route to local network 10.20.30.0/24 via 172.17.0.1 dev eth0
Tue Aug 16 21:42:12 2022 OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
Tue Aug 16 21:42:12 2022 library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
Tue Aug 16 21:42:12 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:42:12 2022 CRL: loaded 1 CRLs from file [[INLINE]]
Tue Aug 16 21:42:12 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:12 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:12 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:42:12 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:42:17 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:42:17 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:17 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:17 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:42:17 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:42:22 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:42:22 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:22 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:22 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:42:22 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:42:27 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:42:27 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:27 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:27 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:42:27 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:42:32 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:42:32 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:32 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:32 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:42:32 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:42:42 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:42:42 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:42 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:42:42 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:42:42 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:43:02 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:43:02 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:43:02 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:43:02 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:43:02 2022 SIGUSR1[soft,init_instance] received, process restarting
Tue Aug 16 21:43:42 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Aug 16 21:43:42 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:43:42 2022 RESOLVE: Cannot resolve host address: us3.privacy.network:1198 (Name does not resolve)
Tue Aug 16 21:43:42 2022 Could not determine IPv4/IPv6 protocol
Tue Aug 16 21:43:42 2022 SIGUSR1[soft,init_instance] received, process restarting

**Reply by adamh98** [reply]
@pkishino Thanks for taking the time to go over this, but do you have any other insight as to what the issue might be?

**Reply by adamh98** [reply]
I'm still not able to figure it out. If anyone has some ideas that would be much appreciated.
